modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_strURLEncode.java

```diff
@@ -0,0 +1,62 @@
+package org.geotools.filter.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+
+/**
+ * URL encodes a string.
+ * This is intended to encode URL paths as such, spaces are encoded as %20, not +
+ *
+ * @see URLEncoder#encode(String)
+ *
+ * @author Billy Newman (BIT Systems)
+ */
+public class FilterFunction_strURLEncode extends FunctionExpressionImpl {
+
+    /**
+     * The FunctionName
+     */
+    public static FunctionName NAME = new FunctionNameImpl("strURLEncode", String.class,
+            parameter("encodeable", String.class));
+
+    /**
+     * Create a new FilterFunction_strURLEncode instance
+     */
+    public FilterFunction_strURLEncode() {
+        super(NAME);
+    }
+
+    /**
+     * URL encode the string.
+     *
+     * @param feature
+     * @return The URL encoded string
+     * @throws IllegalArgumentException if the argument is not a string
+     */
+    @Override
+    public Object evaluate(Object feature) {
+        String arg0;
```

Comment:
Can this variable be renamed? Even if it's just something as simple as "stringToBeEncoded"? Arg0 isn't very informative :)